components/eventing-controller/controllers/backend/reconciler.go

```diff
@@ -703,23 +713,54 @@ func (r *Reconciler) getCurrentBackendCR(ctx context.Context) (*eventingv1alpha1
 	return backend, err
 }
 
-func (r *Reconciler) getOAuth2ClientCredentials(ctx context.Context, name, namespace string) (clientID, clientSecret []byte, err error) {
+func (r *Reconciler) getOAuth2SecretNamespacedName() types.NamespacedName {
+	var name, namespace string
+	if featureflags.IsEventingWebhookAuthEnabled() {
+		name = r.cfg.EventingWebhookAuthSecretName
+		namespace = r.cfg.EventingWebhookAuthSecretNamespace
+	} else {
+		name = getOAuth2ClientSecretName()
+		namespace = deployment.ControllerNamespace
+	}
+	return types.NamespacedName{Name: name, Namespace: namespace}
+}
+
+func (r *Reconciler) getOAuth2ClientCredentials(ctx context.Context) ([]byte, []byte, []byte, error) {
+	var err error
+	var exists bool
+	var clientID, clientSecret, tokenURL []byte
+
 	oauth2Secret := new(v1.Secret)
-	oauth2SecretNamespacedName := types.NamespacedName{Namespace: namespace, Name: name}
+	oauth2SecretNamespacedName := r.getOAuth2SecretNamespacedName()
+
+	r.namedLogger().Infof("Reading secret %s", oauth2SecretNamespacedName.String())
+
 	if getErr := r.Get(ctx, oauth2SecretNamespacedName, oauth2Secret); getErr != nil {
-		err = errors.Wrapf(getErr, "get secret failed namespace:%s name:%s", namespace, name)
-		return
+		err = errors.Wrapf(getErr, "get secret failed namespace:%s name:%s",
+			oauth2SecretNamespacedName.Namespace, oauth2SecretNamespacedName.Name)
+		return clientID, clientSecret, tokenURL, err
 	}
-	var exists bool
-	if clientID, exists = oauth2Secret.Data["client_id"]; !exists {
-		err = errors.New("key 'client_id' not found in secret " + oauth2SecretNamespacedName.String())
-		return
+	if clientID, exists = oauth2Secret.Data[secretKeyClientID]; !exists {
+		err = errors.Errorf("key '%s' not found in secret %s",
+			secretKeyClientID, oauth2SecretNamespacedName.String())
+		return clientID, clientSecret, tokenURL, err
+	}
+	if clientSecret, exists = oauth2Secret.Data[secretKeyClientSecret]; !exists {
+		err = errors.Errorf("key '%s' not found in secret %s",
+			secretKeyClientSecret, oauth2SecretNamespacedName.String())
```

Comment:
we shouldn't log secret data

Reply:
We don't, it is only the secret name and namespace.